kolibri/core/content/generate_fixtures.py

```diff
@@ -0,0 +1,299 @@
+license_owner="ABC org"
+developer = "bedo khaled"
+
+
+
+def GenerateChannel(root_node):
+    
+    channel= ChannelMetadata.objects.create(
+
+        id = generate_random_id(),
+        name =  "main testing channel",
+        author = developer,
+        min_schema_version = "1",
+        root = root_node)
+
+    generated_objects.add(channel)
+    return channel 
+
+
+def GenerateOneContentNode(channel_id,kind,title,description=None,parent=None,available=True,lang_id="en",is_root=False):
+    
+    new_node = ContentNode.objects.create(
+    
+    id = generate_random_id(),
+    parent = parent,
+    channel_id = channel_id,
+    content_id = generate_random_id(),
+    kind=kind,
+
+    license_name=license_name,
+    license_description=license_description,
+    title=title ,
+    description=description,
+    license_owner=license_owner,
+    author = developer,
+    available = available)
+
+    generated_objects.add(new_node)
+    
+    
+    return new_node
+
+
+def GenerateTag(tag_name="test_tag"):
+    tag = ContentTag.objects.create(tag_name=tag_name, id = generate_random_id())
+    generated_objects.add(tag)
+    return tag
+
+
+def GenerateLocalFile(file_preset):
+    # see GetAllLocalFiles() above first 
+    
+    file_extension = file_preset_to_localfile_extension[file_preset]
+
+    # file that will be used is  last one by default
+    file_to_use = AllLocalFiles[file_extension][-1]
+
+    # delete last file (aka used file)
+    AllLocalFiles[file_extension].pop()
+
+    local_file = LocalFile.objects.create(
+        id=file_to_use.id,
+        available = file_to_use.available,
+        file_size = file_to_use.file_size,
+        extension = file_extension) 
+
+    generated_objects.add(local_file)
+    
+    # save for consecutive uses
+    Used_Local_Files[local_file.id] = local_file
+    return local_file
+    
+
+def GenerateFile(contentnode,number_of_files_to_generate):
+    
+    # still didn't figure out the right cases where many file objects cam map to a single localfile
+    # in that case we won't need to generate a new localfile but anyway.
+
+    preset_options = content_kind_to_file_preset[contentnode.kind]
+
+    file_preset = preset_options[random.randint(0,len(preset_options)-1)]
+    local_file = GenerateLocalFile(file_preset)
+
+    file = File.objects.create(
+        id=generate_random_id(),
+        local_file = local_file,
+        contentnode = contentnode,
+        lang = contentnode.lang,
+        preset = file_preset )  
+
+    generated_objects.add(file)
+    return file
+
+ 
+
+def ToMemory():
+    for db in settings.DATABASES:
+
+        settings.DATABASES[db] = {
+            "ENGINE" : "django.db.backends.sqlite3",
+            "NAME"   : ":memory:"
+        }
+        try :
+            del connections[db]
+        except AttributeError : pass 
+        call_command("migrate", interactive=False, database=db)
+
+def StartGenerating():
+
+    GetAllLanguages()
```

Comment:
this functions gets all the languages of the QA channel but i can't use them directly, why? because these languages exist in the local db (not the memory db) and i'm only dumping models generated in the memory, so if used them directly in the nodes for example they will not get dumped in the fixtures.